### PR TITLE
Fix state clearing in tests, change migration resource implementation

### PR DESCRIPTION
**Pull request by adrns**
Always erroring on delete for the migration resource is near impossible to test. There's no supported way in terraform-plugins to test this edge-case behavior. However, I realized that deleting a single migration resource will likely never happen: deletes will occur most of the time during box deletion, so it's perfectly sensible to let users delete a migration which will go away anyway with the box. If that's not the case, there's a warning message printed after deletion.

With deletion being enabled, ForceNew no longer makes sense. So I added an update, which always errors like delete did before with a message.

Added/changed tests accordingly.

As for the original problem of the ticket:
Migration testing leaves some stuff in the db. I wrote a post-test script which just removes migrations from the box.

### CODE REVIEW

File: internal/provider/resource_db_migration.go [hunk 1]
```diff
@@ -55,27 +57,37 @@ func resourceDbMigrationRead(ctx context.Context, data *schema.ResourceData, met
...
-// to do is impossible - as opposed to silently doing nothing here, which could
-// possibly make users think they deleted the migration.
+// There's no such thing as "updating/deleting a migration". However, this doesn't
+// match with terraform's resource model, so we must provide these methods.
+// Trying to update will always fail with printing the below instructions to users.
+// Trying to delete will always succeed, but users get a warning that deleting
+// will leave some state behind in the box - as opposed to silently doing nothing
+// here, which could possibly make users think they deleted the migration.
+// This is OK since most of the time this deletion will occur during the removal
+// of a box (i.e. not by deleting this specific resource).
```
Comment (MFAshby) [inline internal/provider/resource_db_migration.go:67]: :+1: lovely comment thanks

**Review by MFAshby — APPROVED** [review]
lgtm thanks

**Review by k9t5h — APPROVED** [review]
LGTM
